inject/src/main/java/io/micronaut/inject/annotation/AbstractAnnotationMetadataBuilder.java

```diff
@@ -215,7 +215,7 @@ public AnnotationMetadata buildDeclared(T element, List<? extends A> annotations
      * @return The {@link AnnotationMetadata}
      */
     public CachedAnnotationMetadata lookupOrBuildForParameter(T owningType, T methodElement, T parameterElement) {
-        return lookupOrBuild(true, owningType, methodElement, parameterElement);
+        return lookupOrBuild(false, owningType, methodElement, parameterElement);
```

Comment:
The correct value is false. It isn't implemented for parameters anyway. That's why `true` didn't do anything.